### PR TITLE
@zephraph => [Danger] Warn when MP PR introduces breaking changes to V2 schema

### PR DESCRIPTION
**Pull request by mzikherman**
(when compared to currently deployed Reaction).

I _think_ this makes sense as a check? It's sort of the Force production schema check (does MP production have any breaking changes when compared to the one included in the to-be-released version), but inverted.

Does this MP PR have any breaking changes when compared to the currently deployed Force version? If so, that makes Force undeployable.

### CODE REVIEW

**Comment by zephraph** [comment]
So this looks great... but when does it run? I don't _think_ the danger file is being run right now (unless peril is doing it).

**Comment by mzikherman** [comment]
I'm really not sure...there's a `dangerfile.ts` in the root here, so I assumed it _was_? I'm frequently a little unsure exactly _where_ to put a snippet of CI/automation!

**Comment by zephraph** [comment]
So the `schema-drift` file (which I don't think is really actually working) is configured to run here: https://github.com/artsy/metaphysics/blob/e6aac56621e3be4ffe17efa1fb7a0ae80ece587e/.circleci/config.yml#L44

If you want the actual danger file to run, you'd have to add something similar. Peril can run danger files (like this [volt](https://github.com/artsy/peril-settings/blob/master/peril.settings.json#L35-L37)  example), but it only supports certain dependencies so I don't think it'd run this.  

Probably just need to add something to circle.

**Comment by mzikherman** [comment]
Interesting! In that case, I'll refactor to _remove_ this from the Dangerfile (and shall I delete the Dangerfile since it's not used?), and have it as a stand-alone script.

Then, it can run as its own step in the Circle CI flow, just on PR's.

**Comment by zephraph** [comment]
Actually @mzikherman, I think we should just wire up the danger file to be ran 😄

**Comment by mzikherman** [comment]
I actually had commented out the entire dangerfile _besides_ my additions because it was failing :(

I just assumed it was...whatever, no big deal. I just wanted to get a small code win out after a week of struggles 😄 

That makes sense now since it wasn't being ran at all!

**Comment by zephraph** [comment]
Okay! Then I'm 👍 for you to move it out.

**Comment by mzikherman** [comment]
Added a ticket for hooking up the Danger run: https://artsyproduct.atlassian.net/browse/PLATFORM-2161

File: scripts/compare-reaction-schema.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { readFileSync } from "fs"
+const { diff: schemaDiff } = require("@graphql-inspector/core")
+import fetch from "node-fetch"
+
+// If there is a breaking change between the local schema,
+// and the current Reaction one, warn.
+;(async () => {
+  const forcePackageJSON = await (await fetch(
+    "https://raw.githubusercontent.com/artsy/force/release/package.json"
+  )).json()
```
Comment (zephraph) [inline scripts/compare-reaction-schema.ts]: The call to `json()` _shouldn't_ return a promise. I think all you need to do is `(await fetch("...")).json()`

File: scripts/compare-reaction-schema.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  const reactionVersion = forcePackageJSON["dependencies"]["@artsy/reaction"]
+  const reactionSchemaUrl = `https://github.com/artsy/reaction/raw/v${reactionVersion}/data/schema.graphql`
+
+  const reactionSchema = await (await fetch(reactionSchemaUrl)).text()
```
Comment (zephraph) [inline scripts/compare-reaction-schema.ts]: Same here

**Pull request merged**